app/Http/Controllers/HarvestController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Repositories\HarvestRepositoryEloquent;
+use App\Http\Requests\CreateHarvestRequest;
+use App\Http\Requests\UpdateHarvestRequest;
+use App\Http\Resources\HarvestResource;
+use App\Services\HarvestService;
+
+class HarvestController extends Controller
+{
+    protected $harvestRepository;
+
+    public function __construct(HarvestRepositoryEloquent $harvestRepository)
+    {
+        $this->harvestRepository = $harvestRepository;
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $this->harvestRepository->pushCriteria(app('Prettus\Repository\Criteria\RequestCriteria'));
+        $harvests = $this->harvestRepository->all();
+
+        return HarvestResource::collection($harvests);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CreateHarvestRequest $request, HarvestService $harvestService)
+    {
+        $data   = $harvestService->create($request);
+        $harvest = $this->harvestRepository->create($data);
+
+        return new HarvestResource($harvest);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        return new HarvestResource($this->harvestRepository->find($id));
```

Comment:
biar seragam dengan create, gimana kalau sebaiknya dibuat gini ?
$harvest = $this->harvestRepository->find($id);
return new HarvestResource($harvest);